src/main/scala/beam/utils/google_routes_db/ReqRespMapping.scala

```diff
@@ -0,0 +1,210 @@
+package beam.utils.google_routes_db
+
+import akka.Done
+import akka.actor.ActorSystem
+import akka.http.scaladsl.common.{EntityStreamingSupport, JsonEntityStreamingSupport}
+import akka.stream.scaladsl._
+import beam.sim.BeamHelper
+import beam.sim.common.SimpleGeoUtils
+import beam.utils.csv.CsvWriter
+import beam.utils.google_routes_db.config.GoogleRoutesDBConfig
+import beam.utils.google_routes_db.{request ⇒ req, response ⇒ resp}
+import org.matsim.api.core.{v01 ⇒ matsim}
+
+import scala.collection.mutable.ArrayBuffer
+import scala.collection.{immutable, mutable}
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+/**
+ * Output file: ReqRespMapping.csv
+ *
+ * Run with Gradle:
+ * ./gradlew :execute \
+ *   -PmainClass=beam.utils.google_routes_db.ReqRespMapping \
+ *   -PappArgs="['--config','src/main/scala/beam/utils/google_routes_db/config/google_routes_db.conf']" \
+ *   -PlogbackCfg=logback.xml
+ */
+object ReqRespMapping extends BeamHelper {
+
+  private implicit val system: ActorSystem = ActorSystem("google-routes-db")
+  private implicit val execCtx: ExecutionContext = system.dispatcher
+  private implicit val jsonStreamingSupport: JsonEntityStreamingSupport = EntityStreamingSupport.json()
+
+  private var csvWriter: CsvWriter = _
+
+  //noinspection DuplicatedCode
+  def main(args: Array[String]): Unit = try {
+    val (_, cfg) = prepareConfig(args, isConfigArgRequired = true)
+    val config = GoogleRoutesDBConfig(cfg)
+    val geoUtils = SimpleGeoUtils()
+
+    val epsilons = immutable.Seq(0.001, 0.0001, 0.00001)
+    val maxSizeKey = 7
+
+    // googleapiResponsesJsonFileUri,epsilon,maxSizeKey,reqReqSize=0,reqReqSize=1,...
+    val header = ArrayBuffer("googleapiResponsesJsonFileUri", "epsilon", "maxSizeKey")
+    header.appendAll(
+      for {
+        mapping ← Seq("reqReqSize", "reqRespSize", "respReqSize", "respRespSize")
+        i ← 0 to maxSizeKey
+      } yield {
+        s"$mapping=$i"
+      }
+    )
+
+    csvWriter = new CsvWriter(s"ReqRespMapping.csv", header)
+
+    val doneFuture: Future[Done] = sourceGoogleapiFiles(config)
+      .flatMapConcat { googleapiFiles ⇒
+        Source(epsilons).map(epsilon ⇒ (googleapiFiles, epsilon))
+      }
+      .toMat(Sink.foreach { case (googleapiFiles, epsilon) ⇒
+        val requests =
+          req.csv.parseGoogleTravelTimeEstimationCsv(
+            googleapiFiles.googleTravelTimeEstimationCsvText
```

Comment:
Will it download the same files `epsilons.length` times? If yes, should it be cached?

Reply:
I don't think so. `sourceGoogleapiFiles` yields downloaded contents in _xxxText_ fields, i.e _googleTravelTimeEstimationCsvText_, which is parsed here.